src/jit/lower.cpp

```diff
@@ -1881,16 +1729,29 @@ GenTree* Lowering::LowerTailCallViaHelper(GenTreeCall* call, GenTree *callTarget
         InsertPInvokeMethodEpilog(comp->compCurBB DEBUGARG(call));
     }
 
-    // Remove gtCallAddr from execution order if one present.
-    GenTreeStmt* callStmt = comp->compCurStmt->AsStmt();     
+    // Remove gtCallAddr from execution order if present.
+    LIR::Range callTargetRange;
     if (call->gtCallType == CT_INDIRECT)
     {
         assert(call->gtCallAddr != nullptr);
-        comp->fgDeleteTreeFromList(callStmt, call->gtCallAddr);
+
+        bool isClosed;
+        callTargetRange = m_blockRange.GetTreeRange(call->gtCallAddr, &isClosed);
+        (assert(isClosed), (void)isClosed);
+
```

Comment:
As it turns out, we don't need it. The unused var warning that I expected is only enabled in the CoreFX native build.